libs/edssser/edssser.go

```diff
@@ -0,0 +1,117 @@
+package edssser
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+
+	"github.com/celestiaorg/celestia-app/pkg/da"
+
+	"github.com/celestiaorg/celestia-node/share/eds"
+	"github.com/celestiaorg/celestia-node/share/eds/edstest"
+)
+
+type Config struct {
+	EDSSize   int
+	EDSWrites int
+	EnableLog bool
+}
+
+// EDSsser stand for EDS Store Stresser.
+type EDSsser struct {
+	config     Config
+	datastore  datastore.Batching
+	edsstoreMu sync.Mutex
+	edsstore   *eds.Store
+}
+
+func NewEDSsser(path string, datastore datastore.Batching, cfg Config) (*EDSsser, error) {
+	edsstore, err := eds.NewStore(path, datastore)
+	if err != nil {
+		return nil, err
+	}
+
+	return &EDSsser{
+		config:    cfg,
+		datastore: datastore,
+		edsstore:  edsstore,
+	}, nil
+}
+
+func (ss *EDSsser) Run(ctx context.Context) (stats Stats, err error) {
+	ss.edsstoreMu.Lock()
+	defer ss.edsstoreMu.Unlock()
+
+	err = ss.edsstore.Start(ctx)
+	if err != nil {
+		return stats, err
+	}
+	defer func() {
+		err = errors.Join(err, ss.edsstore.Stop(ctx))
+	}()
+
+	t := &testing.T{}
+	for ctx.Err() == nil && ss.config.EDSWrites != 0 {
```

Comment:
I might be missing it, but shouldn't ss.config.EDSWrites be backwards iterated somewhere so `ss.config.EDSWrites != 0 ` becomes a termination condition at some point?

Reply:
No, I missed the decrement somehow

Reply:
[6944037](https://github.com/celestiaorg/celestia-node/pull/2482/commits/6944037041b207e67a2897041c6ea592c25bf5b5)

Reply:
https://github.com/celestiaorg/celestia-node/pull/2482/commits/6944037041b207e67a2897041c6ea592c25bf5b5